src/Joblify.Core.Api/Offers/OfferController.cs

```diff
@@ -60,6 +61,47 @@ public async Task<IActionResult> GetAllTrades()
             return Ok(response);
         }
 
+        [HttpGet("getOffersForUser")]
+        public async Task<IActionResult> GetOffersForUser(int id)
+        {
+            var response = await _offerService.GetOffersForUser(id);
+            if (response == null)
+            {
+                return BadRequest();
+            }
+            return Ok(response);
+        }
+
+        [HttpPut("UpdateOffer")]
+        public async Task<IActionResult> UpdateOffer(int id, [FromBody]EditOfferDto offer)
+        {
+            if (offer is null)
+            {
+                return BadRequest();
+            }
+
+            if (!ModelState.IsValid)
+            {
+                return new UnprocessableEntityObjectResult(ModelState);
+            }
+
+            var offerFromDatabase = await _offerService.GetOfferEntity(id);
+
```

Comment:
maybe it would be better to check it in service not here? can someone comment on that?

Reply:
I think it's ok, bc we do similar thing in many other cases. It will be redundacy if I change it. I know we do our best to treat the code as an art so If I'm wrong please correct me, I can fix this part of code @EnjoyTheNoise @blc132 @Bartosz97 @JJMPdsk

Reply:
@Astenna is right, this is not responsibility of controller. Where do we do it similarly, can you point?

Reply:
I thought  that if we ask for an entity in controller once, we can also ask twice in updateOffer case. I changed it to follow rules of layers but now I hit database one more time than in the previous version.